core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -144,7 +143,7 @@
         Preconditions.checkNotNull(builder.fallbackRegistry, "fallbackRegistry");
     Preconditions.checkNotNull(transportServers, "transportServers");
     Preconditions.checkArgument(!transportServers.isEmpty(), "no servers provided");
-    this.transportServers = new ArrayList<>(transportServers);
+    this.transportServer = transportServers.get(0);
```

Comment:
Because this is a two way door: it allows starting multiple servers, each supports multiple ports. And at that time, it won't break users' code again. Do you suggest to pass in only one server?

Reply:
I don't understand. Only the first server is accessed. How would code not be broken if they passed two servers?

FWIW, this is not directly called by users, but instead is called by the builder.

Reply:
Because it is called from builder and Netty is the only place right now we provide this list, and we guarantee it is a singleton list. So there won't be two servers passing in right now. 
What I meant was that, IIRC, we might consider supporting multiple transportServers in the ServerImpl, each of those transport servers can support multiple ports, and at that time, we would need a list of transportServers here in the constructor, instead of one object, and if we keep it as a list, we do not need refactor again at that time.

I agree the "singleton list guarantee" is less intuitive and can be confusing, as a trade off of maybe forward compatibility. I do not have strong preference over one or another, and It looks like both of you prefer passing in a single object. I can refactor it then.